packages/delegation-controller/src/DelegationController.ts

```diff
@@ -0,0 +1,274 @@
+import type { StateMetadata } from '@metamask/base-controller';
+import { BaseController } from '@metamask/base-controller';
+import { SignTypedDataVersion } from '@metamask/keyring-controller';
+import { hexToNumber } from '@metamask/utils';
+
+import { ROOT_AUTHORITY } from './constants';
+import type {
+  Address,
+  Delegation,
+  DelegationControllerMessenger,
+  DelegationControllerState,
+  DelegationEntry,
+  DelegationFilter,
+  DeleGatorEnvironment,
+  Hex,
+  UnsignedDelegation,
+} from './types';
+import { createTypedMessageParams, isHexEqual } from './utils';
+
+export const controllerName = 'DelegationController';
+
+const delegationControllerMetadata = {
+  delegations: {
+    persist: true,
+    anonymous: false,
+  },
+} satisfies StateMetadata<DelegationControllerState>;
+
+/**
+ * Constructs the default {@link DelegationController} state. This allows
+ * consumers to provide a partial state object when initializing the controller
+ * and also helps in constructing complete state objects for this controller in
+ * tests.
+ *
+ * @returns The default {@link DelegationController} state.
+ */
+function getDefaultDelegationControllerState(): DelegationControllerState {
+  return {
+    delegations: {},
+  };
+}
+
+/**
+ * The {@link DelegationController} class.
+ * This controller is meant to be a centralized place to store and sign delegations.
+ */
+export class DelegationController extends BaseController<
+  typeof controllerName,
+  DelegationControllerState,
+  DelegationControllerMessenger
+> {
+  private readonly hashDelegation: (delegation: Delegation) => Hex;
```

Comment:
We have been conventionally using ES private field syntax (`#foo`) instead of the `private` key as it offers better security, would you mind changing this? e.g.:

``` suggestion
  readonly #hashDelegation: (delegation: Delegation) => Hex;